test/cbmc/stubs/mbedtls_stubs.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * corePKCS11 V2.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file mbedtls_stubs.c
+ * @brief Stubs for mbed TLS functions.
+ */
+
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/threading.h"
+
+
+void mbedtls_entropy_init( mbedtls_entropy_context * ctx )
+{
+    ( void ) ctx;
```

Comment:
Why not `__CPROVER_assert( ctx != NULL )` here?
Stubs should superficially reflect what the actual function is doing.

If by any chance `ctx == NULL`, then [this line](https://github.com/ARMmbed/mbedtls/blob/848a4e06b375e067552f1a21d4bc69322c673217/library/entropy.c#L92) would give a Segmentation Fault thus violating the memory safety.

I believe same would apply to all the stubs below.

Reply:
Good point, I have added this